app/scripts/modules/amazon/src/loadBalancer/configure/choice/awsLoadBalancerChoice.modal.ts

```diff
@@ -0,0 +1,47 @@
+import { IComponentController, module } from 'angular';
+
+import { Application } from '@spinnaker/core';
+
+import { IAwsLoadBalancerConfig, LoadBalancerTypes } from './LoadBalancerTypes';
+
+import './awsLoadBalancerChoice.less'
+
+class AwsLoadBalancerChoiceCtrl implements IComponentController {
+  public choices: IAwsLoadBalancerConfig[];
+  public choice: IAwsLoadBalancerConfig;
+
+  constructor(public $uibModal: any,
+              public $uibModalInstance: any,
+              private application: Application) {
+    'ngInject';
+  }
+
+  public $onInit (): void {
+    this.choices = LoadBalancerTypes;
+    this.choice = this.choices[0];
+  }
+
+  public onChoiceSelection (choice: IAwsLoadBalancerConfig): void {
+    this.choice = choice;
+  }
+
+  public choose (): void {
+    this.$uibModalInstance.dismiss();
+    this.$uibModal.open({
```

Comment:
UI bootstrap modal has a pretty nice component and promise API, but since we're migrating to React, I'm not going to mention it.